src/entities/createPool/createPoolV3/gyroECLP/createPoolGyroECLP.ts

```diff
@@ -85,3 +86,75 @@ export class CreatePoolGyroECLP implements CreatePoolBase {
         });
     }
 }
+
+export function calcDerivedParams(params: EclpParams): DerivedEclpParams {
+    const D18 = 10n ** 18n; // 18 decimal precision is how params are stored
+    const D38 = 10n ** 38n; // 38 decimal precision for derived param return values
+    const D100 = 10n ** 100n; // 100 decimal precision for internal calculations
+
+    let { alpha, beta, c, s, lambda } = params; // params start at 18
+
+    // scale from 18 to 100
+    c = (c * D100) / D18;
+    s = (s * D100) / D18;
+    lambda = (lambda * D100) / D18;
+    alpha = (alpha * D100) / D18;
+    beta = (beta * D100) / D18;
+
+    const dSq = (c * c + s * s) / D100; // divide by D100 to keep at 100 decimal precision
+
+    // const d = GyroPoolMath.sqrt(dSq, 5n); // sir isaac style throws error _sqrt FAILED
+    const d = bigIntSqrt(dSq) * 10n ** 50n; // square root reduces precision to 50 decimal places?
+
+    // dAlpha and dBeta result in 53 decimal precision?
+    const dAlpha =
+        D100 /
+        bigIntSqrt(
+            (((c * D100) / d + (alpha * s) / d) ** 2n * D100) / lambda ** 2n +
+                ((alpha * c) / d - (s * D100) / d) ** 2n / D100,
+        );
+    const dBeta =
+        D100 /
+        bigIntSqrt(
+            (((c * D100) / d + (beta * s) / d) ** 2n * D100) / lambda ** 2n +
+                ((beta * c) / d - (s * D100) / d) ** 2n / D100,
+        );
+
+    const tauAlpha: Vector2 = {
+        // weird 50 scaling necessary to get back to 100 because of dAlpha precision?
+        x: (((alpha * c) / D100 - s) * dAlpha) / 10n ** 50n,
+        y: ((((c + (s * alpha) / D100) * dAlpha) / 10n ** 50n) * D100) / lambda,
+    };
+    const tauBeta: Vector2 = {
+        // weird 50 scaling necessary to get back to 100 because of dBeta precision?
+        x: (((beta * c) / D100 - s) * dBeta) / 10n ** 50n,
+        y: ((((c + (s * beta) / D100) * dBeta) / 10n ** 50n) * D100) / lambda,
+    };
+
+    // Each multiplication must be scaled down by D100
+    const w = (s * c * (tauBeta.y - tauAlpha.y)) / (D100 * D100);
+    const z = (c * c * tauBeta.x + s * s * tauAlpha.x) / (D100 * D100);
+    const u = (s * c * (tauBeta.x - tauAlpha.x)) / (D100 * D100);
+    const v = (s * s * tauBeta.y + c * c * tauAlpha.y) / (D100 * D100);
+
+    // all return values scaled to 38 decimal places
+    return {
+        tauAlpha: {
+            x: (tauAlpha.x * D38) / D100,
+            y: (tauAlpha.y * D38) / D100,
+        },
+        tauBeta: {
+            x: (tauBeta.x * D38) / D100,
+            y: (tauBeta.y * D38) / D100,
+        },
+        u: (u * D38) / D100,
+        v: (v * D38) / D100,
+        w: (w * D38) / D100,
+        z: (z * D38) / D100,
+        dSq: (dSq * D38) / D100,
+    };
+}
+
+function bigIntSqrt(val: bigint): bigint {
+    return BigInt(new Big(val.toString()).sqrt().toFixed(0));
```

Comment:
Would the existing [gyro math sqrt implementation](https://github.com/balancer/balancer-maths/blob/main/typescript/src/gyro/gyroPoolMath.ts#L15) work?

Reply:
I tried to use that sir isaac newton style sqrt first, but it runs into the `throw Error('_sqrt FAILED')` when feeding a 100 decimal number as the input with any sane amount of tolerance. ( it does succeed with tolerance `1000000000000000000000000000000000000000000000000000000000000n` :joy: )

But our AI friends think that it is theoretically possible to modify that `GyroPoolMath.sqrt` function to work with 1e100 scale :thinking:

Reply:
Definitely don't want to modify `GyroPoolMath.sqrt` as it needs to match the smart contract implementation.
Looks like we will live with the dependencie addition.